### ISSUE TITLE
Display model elements in a tabular format

### ISSUE THREAD

**Issue by sz332** [gaphor/gaphor#2568]
### Is your feature request related to a problem?

As a modeler, I look at the model as a database, therefore:
- I would like to query the model and display the results in a tabular format because it's the most common way to see such information
- I would like to do the query without (much/any) "programming knowledge" because most modelers are not software engineers. 
- Although I would like to do the query as simple as possible being able to write complex query provides me value because then I don't need to rely on 3rd party tools to refine the output I can generate from a modeling tool
- I would like to be able to export the result into a standardized format (for example CSV) in order to use to be able to process the result with a more appropriate software (eg. Excel)
- I would like to configure what is being displayed (what elements, what attributes, etc.) because I only what to see what is relevant for me
- I would like to configure which part of the model is being used to generate the result (for example all elements under a package, elements of type X) because I often need only to know about certain data (requirements for X system stored in Z package and below)
- I would like to be able to filter the result in order to quickly refine the result and look up things without going through all the results one by one
- I would like to go through every element even if there is more than my screen can display because often the number of elements is so much that they don't fit a single screen
- I would like to be able to jump to a selected result in the model navigation tree in order to use them for further operations (rename, modification, etc.)

**Comment by mikekidner** [comment]
This is a good idea, but I think it should be realised as a plugin, so as not to distract from the clean interface Gaphor currently has.

**Comment by sz332** [comment]
In other modeling tools this is basically a diagram type.

**Comment by tompkins-ct** [comment]
As for requirements, relationships can be extracted in the Python interface pretty easily for export to whatever format makes sense. You could tool it to perform both import/export operations with a plugin to allow for external review/editing to push back into the model.

**Comment by sz332** [comment]
I would argue that having a tabular format would be useful not for only requirements but all sort of model elements. We use this feature of another modeling tool pretty often.

I can say however that the ability to import /export requirements in a standard (for example ReqIF) format could be useful, and if you wish, we could create a separate ticket to discuss it.

**Comment by mikekidner** [comment]
Yep, tabular format would be handy. I was thinking more about import/export auto generation type of thing. 
With tabular diagrams, do you imagine a basic grid where users can pick headers for each row and column from a drop down/filter and then the selected relations auto fill the grid? 

<img width="349" alt="Screen Shot 2023-07-29 at 8 35 31 am" src="https://github.com/gaphor/gaphor/assets/50702106/b38c909a-a36b-4342-a454-ea2973f23bb3">

This would work well for s few elements if the diagram is trying to show something specific.
When the number of elements gets large ie  (> 10) that process and diagram layout would be cumbersome, which is where plugin tools to export requirements coverage table (or other model query) to csv directly would be handy. 
The plugin could handle the filtering/query interface in a more extensive way.
I agree that users shouldn't be expected to write python just to pull a standard view from the model.

**Comment by sz332** [comment]
@mikekidner I created a separate ticket for the grid feature which I think the describes something very similar that you suggested. Can you take a look at this: #2569

**Comment by mikekidner** [comment]
See also #927
